cmd/wallet/deposits.go

```diff
@@ -0,0 +1,189 @@
+			return deposits[i].amountBtc < deposits[j].amountBtc
+		})
+	}
+
+	// Print
+	if err := printTable(deposits); err != nil {
+		return fmt.Errorf("failed to print deposits table: %v", err)
+	}
+
+	return nil
+}
+
+func removeSwept(deposits []depositEntry) []depositEntry {
+	result := []depositEntry{}
+	for _, deposit := range deposits {
+		if deposit.isSwept {
+			continue
+		}
+		result = append(result, deposit)
+	}
+	return result
+}
+
+func getDeposits(tbtcChain tbtc.Chain, walletStr string) ([]depositEntry, error) {
+	logger.Infof("reading deposits from chain...")
+
+	filter := &tbtc.DepositRevealedEventFilter{}
+	if len(walletStr) > 0 {
+		walletPublicKeyHash, err := hexToWalletPublicKeyHash(walletStr)
+		if err != nil {
+			return []depositEntry{}, fmt.Errorf("failed to extract wallet public key hash: %v", err)
+		}
+
+		filter.WalletPublicKeyHash = [][20]byte{walletPublicKeyHash}
+	}
+
+	depositRevealedEvent, err := tbtcChain.PastDepositRevealedEvents(filter)
+	if err != nil {
+		return []depositEntry{}, fmt.Errorf(
+			"failed to get past deposit revealed events: [%w]",
+			err,
+		)
+	}
+
+	logger.Infof("found %d DepositRevealed events", len(depositRevealedEvent))
+
+	result := make([]depositEntry, len(depositRevealedEvent))
+	for i, event := range depositRevealedEvent {
+		logger.Debugf("getting details of deposit %d/%d", i+1, len(depositRevealedEvent))
+
+		depositKey := buildDepositKey(event.FundingTxHash, event.FundingOutputIndex)
+
+		depositRequest, err := tbtcChain.GetDepositRequest(event.FundingTxHash, event.FundingOutputIndex)
+		if err != nil {
+			return result, fmt.Errorf(
+				"failed to get deposit request: [%w]",
+				err,
+			)
+		}
+
+		result[i] = depositEntry{
+			wallet:                        event.WalletPublicKeyHash,
+			depositKey:                    depositKey,
+			revealedAtBlock:               event.BlockNumber,
+			isSwept:                       depositRequest.SweptAt.After(depositRequest.RevealedAt),
+			fundingTransactionHash:        event.FundingTxHash,
+			fundingTransactionOutputIndex: event.FundingOutputIndex,
+			amountBtc:                     convertSatToBtc(float64(depositRequest.Amount)),
+		}
+	}
+
+	return result, nil
+}
+
+func printTable(deposits []depositEntry) error {
+	w := tabwriter.NewWriter(os.Stdout, 2, 4, 1, ' ', tabwriter.AlignRight)
+	fmt.Fprintf(w, "index\twallet\tvalue (BTC)\tdeposit key\tfunding transaction\tswept\t\n")
+
+	for i, deposit := range deposits {
+		fmt.Fprintf(w, "%d\t%s\t%.5f\t%s\t%s\t%v\t\n",
+			i,
+			"0x"+hex.EncodeToString(deposit.wallet[:]),
+			deposit.amountBtc,
+			deposit.depositKey,
+			fmt.Sprintf(
+				"%s:%d",
+				deposit.fundingTransactionHash.Hex(bitcoin.ReversedByteOrder),
+				deposit.fundingTransactionOutputIndex,
+			),
+			deposit.isSwept,
+		)
+	}
+
+	if err := w.Flush(); err != nil {
+		return fmt.Errorf("failed to flush the writer: %v", err)
+	}
+
+	return nil
+}
+
+func hexToWalletPublicKeyHash(str string) ([20]byte, error) {
```

Comment:
Feels like something we could place in `pkg/bitcoin` and import from here. Thoughts on the best location for this logic @lukasz-zimnoch?

Reply:
Not sure if `pkg/bitcoin` is the right place as it should only contain components specific to the Bitcoin protocol. This function is not such a component. It's rather a helper specific to `cmd` package and IMO, should remain as is.

Reply:
I looked at this function one more time, it's just `string` -> `[20]byte` conversion with no hashing magic. Let's leave it as is. It was late in the evening and I was wrong.